src/main/java/com/vinylteam/vinyl/service/JwtTokenProvider.java

```diff
@@ -16,24 +23,40 @@
 import javax.annotation.PostConstruct;
 import javax.crypto.SecretKey;
 import javax.servlet.http.HttpServletRequest;
-import java.util.Collection;
-import java.util.Date;
-import java.util.Objects;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
 
 import static com.vinylteam.vinyl.security.SecurityConstants.AUTHORIZATION_HEADER_NAME;
 import static com.vinylteam.vinyl.security.SecurityConstants.TOKEN_PREFIX;
 
 @Service
+@RequiredArgsConstructor
 @Slf4j
 public class JwtTokenProvider implements JwtService {
 
+    private AuthenticationManager authenticationManager;
+
+    private final UserService userService;
+    private final UserMapper userMapper;
     private SecretKey secretKey;
 
-    @Value("${jwt.token.expired:120000}")
-    private int validityInMilliseconds;
+    @Autowired
+    public void setTokenStorageService(LogoutTokenStorageService tokenStorageService) {
```

Comment:
Why setter autowiring not field one ?

Reply:
it is to avoid cyclic references in the app`s Spring configuration.
Could you please advise is it a good practice to use autowired setters?

Reply:
According to spring documentation: 
`Since you can mix constructor-based and setter-based DI, it is a good rule of thumb to use constructors for mandatory dependencies and setter methods or configuration methods for optional dependencies`

Reply:
we will work with it in the next spring

https://app.clickup.com/t/u3xuc6